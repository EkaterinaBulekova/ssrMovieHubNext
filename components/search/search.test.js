import React from 'react';
import { shallow } from 'enzyme';
import Search from './search';
import toJson from 'enzyme-to-json';

describe('<Search />', () => {
  it('should render as expect', () => {
    const testOnClick = jest.fn(); 
    const component = shallow(<Search/>);
//    const filterButtons = component.find('button.search-filter-button');
    // const form = component.find('form');
    // component.instance().onSearchByClick = jest.fn();
    // component.update();

    expect(toJson(component)).toMatchSnapshot();

    // expect(component.find('div.search-field-title').text()).toBe('FIND YOUR MOVIE');
    // const event = {
    //   target:[
    //     {value: 'test'}
    //   ]
    // }

    // form.simulate('submit', event);
    // expect(testOnClick).toBeCalled();
    // expect(testOnClick).toBeCalledWith({search:'test', searchBy: 'title'});

    // expect(component.find('div.search-filter-title').text()).toBe('SEARCH BY');
    // expect(filterButtons).toHaveLength(2);
    // filterButtons.at('1').simulate('click');
    // expect(component.instance().onSearchByClick).toBeCalled();
    // expect(component.instance().onSearchByClick).toBeCalledWith('genres');

    component.unmount();
  });
});