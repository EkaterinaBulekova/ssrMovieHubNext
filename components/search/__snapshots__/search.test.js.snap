// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Search /> should render as expect 1`] = `
<ContextConsumer>
  <Component />
</ContextConsumer>
`;
